test/jiraCommon.test.js

```diff
@@ -162,3 +12,80 @@ test("Call to retrieve an issues status", function() {
+      customfield_version_released: {name:"released version",released: true},
+      components: [{name:"component 1"},{name:"component 2"}],
+      fixVersions: [{name:"fix Version 1"}],
+      versions: []
+    }
+  }
+  initJiraDummyConfig();
+  expect(EpicField.isUsable()).toBeTruthy();
+  expect(getCfg('jira_url')).toBe("https://jiraserver");
+  UserStorage.setValue(
+    "favoriteCustomFields",
+    [
+      {key:"customfield_custom1",name:"Custom 1",type: "number"},
+      {key:"customfield_custom2",name:"Custom 2",schemaType: "string"},
+      {key:"customfield_custom3",name:"Custom 3",schemaType: "option"},
+      {key:"customfield_stringArray",name:"String Array",schemaType: "array|string"},
+      {key:"customfield_stringArray2",name:"String Array",schemaType: "array|string"},
+      {key:"customfield_stringArray3",name:"String Array",schemaType: "array|string"},
+      {key:"customfield_versions",name:"Version Array",schemaType: "array|versions"},
+      {key:"customfield_emptyversions",name:"Empty Version Array",schemaType: "array|versions"},
+      {key:"customfield_version_released",name:"Version",schemaType: "versions"},
+      {key:"customfield_version_unreleased",name:"Version",schemaType: "versions"},
+      
+      
+    ]
+  );
+  expect(unifyIssueAttrib("summary",testIssue).value).toBe("A summary");
+  expect(unifyIssueAttrib("description",testIssue).value).toBe("This is the description");
+  expect(unifyIssueAttrib("environment",testIssue).value).toBe("An environment");
+  var epicResult = unifyIssueAttrib("customfield_epic_link",testIssue);
+  expect(epicResult.value).toBe("EPC-22");
+  expect(epicResult.link).toBe("https://jiraserver/browse/EPC-22");
+  expect(unifyIssueAttrib("customfield_custom1",testIssue).value).toBe(22);
+  expect(unifyIssueAttrib("customfield_custom1",testIssue).format).toBe("0");
+  expect(unifyIssueAttrib("customfield_custom2",testIssue).value).toBe("hello");
+  expect(unifyIssueAttrib("customfield_custom3",testIssue).value).toBe("option_value");
+  expect(unifyIssueAttrib("customfield_stringArray",testIssue).value).toBe("one,two,three");
+  expect(unifyIssueAttrib("customfield_stringArray2",testIssue).value).toBe("");
+  expect(unifyIssueAttrib("customfield_stringArray3",testIssue).value).toBe("one");
+  expect(unifyIssueAttrib("customfield_versions",testIssue).value).toBe("version1, version2");
+  expect(unifyIssueAttrib("customfield_emptyversions",testIssue).value).toBe("");
+  expect(unifyIssueAttrib("customfield_version_released",testIssue).value).toBe("released version");
+  expect(unifyIssueAttrib("customfield_version_released",testIssue).format).toBe("@[green]");
+  expect(unifyIssueAttrib("customfield_version_unreleased",testIssue).value).toBe("unreleased version");
+  expect(unifyIssueAttrib("customfield_version_unreleased",testIssue).format).toBe("");
+  
+  expect(unifyIssueAttrib("components",testIssue).value).toBe("component 1, component 2");
```

Comment:
Tests for bugfix #161 - slowly improving coverage in "unifyIssueAttrib" - theres a lot of duplicated code in there which could be refactored one day but it needs good testing to make me (or you) confident to do it.